### PR TITLE
Gcc bgq docu [skip ci]

### PR DESCRIPTION
**Pull request by apeyser**
Replacement for pull request #385 and issue #412 

@heplesser please take a look at it

### CODE REVIEW

**Comment by apeyser** [comment]
Add @gtrensch to look at this

File: INSTALL [hunk 1]
```diff
@@ -133,16 +134,33 @@ Compiling for BlueGene/Q
...
+          -DCMAKE_INSTALL_PREFIX:PATH=</install/path> \
+          -Dwith-python=OFF \
+          -Dstatic-libraries=ON \
           </path/to/NEST/src>
 
-It is recommended to build statically on larger BlueGene/Q systems. If you need
-PyNEST on BlueGene/Q, you have to compile dynamically, i.e. `-Dstatic-libraries=OFF`
-and you have to cythonize the `pynest/pynestkernel.pyx/.pxd`
-on a machine with Cython installed:
+If you compile dynamically, be aware that the BlueGene/Q system might not provide a `ltdl`
```
Comment (jougs) [inline INSTALL]: "an `ltdl` library" instead of "a `ltdl` library"

File: INSTALL [hunk 2]
```diff
@@ -133,16 +134,33 @@ Compiling for BlueGene/Q
...
+library. If you want to dynamically load an external user module, you have to
+compile and install a `ltdl` yourself and add `-Dwith-ltdl=<ltdl-install-dir>`
+to the `cmake` line. Otherwise add `-Dwith-ltdl=OFF`.
+
+Additionally, the design of cmake's mpi handling has a broken design, which is
```
Comment (jougs) [inline INSTALL]: make "mpi" uppercase

File: INSTALL [hunk 3]
```diff
@@ -133,16 +134,33 @@ Compiling for BlueGene/Q
...
+brittle in the case of BGQ and certain libraries (flags to use sionlib, for example).
```
Comment (jougs) [inline INSTALL]: sionlib -> SIONlib

File: INSTALL [hunk 4]
```diff
@@ -133,16 +134,33 @@ Compiling for BlueGene/Q
...
+
+If you run into that, you must force cmake to use the wrappers rather than
```
Comment (jougs) [inline INSTALL]: put "cmake" in backticks

File: INSTALL [hunk 5]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
+if they are not found OR the incorrect libraries are found, e.g.:
 
     -DPYTHON_LIBRARY=/bgsys/tools/Python-2.7/lib64/libpython2.7.so.1.0
     -DPYTHON_INCLUDE_DIR=/bgsys/tools/Python-2.7/include/python2.7
+or (much better)
+    -DPYTHON_LIBRARY=/bgsys/local/python3/3.4.2/lib/libpython3.4m.a
+    -DPYTHON_INCLUDE_DIR=/bgsys/local/python3/3.4.2/include/python3.4m
+
+A complete `cmake` line for PyNEST could look like this:
+
```
Comment (jougs) [inline INSTALL]: \`\`\`

File: INSTALL [hunk 6]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
+          -DCMAKE_INSTALL_PREFIX=</install/path> \
+          -Dstatic-libraries=OFF \
+          -Dcythonize-pynest=OFF \
+	  -DCMAKE_C_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlc_r \
+	  -DCMAKE_CXX_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlcxx_r \
+	  -Dwith-python=/bgsys/local/python3/3.4.2/bin/python3 \
+	  -DPYTHON_LIBRARY=/bgsys/local/python3/3.4.2/lib/libpython3.4m.a \
+	  -DPYTHON_INCLUDE_DIR=/bgsys/local/python3/3.4.2/include/python3.4m \
+          -Dwith-ltdl=OFF \
+          <nest-src>
```
Comment (jougs) [inline INSTALL]: \`\`\`

File: INSTALL [hunk 7]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
 
 Further, for running PyNEST, make sure all python dependencies are installed and
 environment variables are set properly:
```
Comment (jougs) [inline INSTALL]: \`\`\`

File: INSTALL [hunk 8]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
-    runjob --exp-env HOME --exp-env PYTHONPATH ... : python script.py
+    runjob \
+    --exp-env HOME \
+    --exp-env PATH \
+    --exp-env LD_LIBRARY_PATH \
+    --exp-env PYTHONUNBUFFERED \
+    --exp-env PYTHONPATH \
+    ... \
+    : /bgsys/local/python3/3.4.2/bin/python3.4 script.py
+
```
Comment (jougs) [inline INSTALL]: \`\`\`

File: INSTALL [hunk 9]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
+BlueGene/Q and GCC
+------------------
+
+Compiling NEST with GCC (`-DCMAKE_TOOLCHAIN_FILE=Platform/BlueGeneQ_GCC`)
+might require you to use a GSL library compiled using GCC, otherwise undefined
+symbols break your build. After the GSL is build with GCC and installed in
```
Comment (jougs) [inline INSTALL]: second "build" -> "built"

File: INSTALL [hunk 10]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
+<gsl-install-dir>, add `-Dwith-gsl=<gsl-install-dir>` to the `cmake` line.
 
 Compiling for Fujitsu Sparc64
```
Comment (jougs) [inline INSTALL]: Please also put the code blocks in this section in triple backticks and remove their indentation

**Comment by jougs** [comment]
Nice work. Please see my minor comments on language and layout.

File: INSTALL [hunk 11]
```diff
@@ -133,16 +134,33 @@ Compiling for BlueGene/Q
...
+
+    -DCMAKE_C_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlc_r
+    -DCMAKE_CXX_COMPILER=/bgsys/drivers/ppcfloor/comm/xl/bin/mpixlcxx_r
+
+BlueGene/Q and PyNEST
+---------------------
+
+Building PyNEST on BlueGene/Q requires you to compile dynamically, i.e. 
+`-Dstatic-libraries=OFF`. Further, you have to cythonize the
+`pynest/pynestkernel.pyx/.pxd` on a machine with Cython installed:
```
Comment (heplesser) [inline INSTALL]: What is the `.pxd` ending?

File: INSTALL [hunk 12]
```diff
@@ -151,20 +169,60 @@ BlueGene/Q and point `-Dwith-python=<...>` to a valid python version for cross
...
     # makes HOME and PYTHONPATH available for python
-    runjob --exp-env HOME --exp-env PYTHONPATH ... : python script.py
+    runjob \
+    --exp-env HOME \
+    --exp-env PATH \
+    --exp-env LD_LIBRARY_PATH \
+    --exp-env PYTHONUNBUFFERED \
+    --exp-env PYTHONPATH \
+    ... \
+    : /bgsys/local/python3/3.4.2/bin/python3.4 script.py
```
Comment (heplesser) [inline INSTALL]: Why the colon at the beginning of the line?

**Comment by heplesser** [comment]
@apeyser I tested the instructions for NEST with XLC and Python 3.4 on Juqueen and they work like a charm!

:+1:

File: INSTALL (hunk 5 above)
Comment (apeyser) [inline INSTALL]: @jougs  So, use github markup? And do this piecewise as we go?

File: INSTALL (hunk 11 above)
Comment (apeyser) [inline INSTALL]: @heplesser `.pxd` is the equivalent of a cython header, and goes with the `.pyx`. Will change that to pyx, for simplicities sake.

File: INSTALL (hunk 5 above)
Comment (jougs) [inline INSTALL]: I assumed so as single backticks were already used in other places of the file. I'd say that the triple ones at least don't hurt readability in a non-markdown enabled environment and help if rendered correctly.

**Comment by apeyser** [comment]
@jougs Can you check the current state?

**Comment by apeyser** [comment]
@jougs : any problem merging and closing?

**Comment by jougs** [comment]
@apeyser: nope. Just skipped my attention. Happy and merging!

**Pull request merged**